geps/gep-1897/index.md

```diff
@@ -5,61 +5,51 @@
-[GRPCRoute](../../reference/spec.md#gateway.networking.k8s.io/v1alpha2.GRPCRoute) use cases
-are not addressed here, because at this point in time these two route types are not graduated to beta.
-2. Mutual TLS (mTLS) use cases are intentionally out of scope for this GEP for two reasons.  First, the design of Gateway
-API is backend-attached and does not currently support mutual authentication, and also because this GEP does not
-address the case where connections to TLS are **implicitly configured** on behalf of the user, which is the norm for mTLS.
-This GEP is about the case where an application developer needs to **explicitly express** that they expect TLS when
-there is no automatic, implicit configuration available.
-3. Service mesh use cases are not addressed here because this GEP is specifically concerned with the connection between
-Gateways and Backends, not Service to Service.  Service mesh use cases should ignore the design components described in
-this proposal.
+2. The gateway implementation (`Gateway` instance or service mesh client proxy) must have final control over security properties
+3. The application developer may provide *hints* or *defaults* to consumers on how to connect to their application.
 
 ## Non-Goals
 
-These are worthy goals, but will not be covered by this GEP.
+A major non-goal for this proposal is for configuration of the service mesh transport security.
+It is very common for service mesh implementations to implement some form of transparent transport security, whether that is WireGuard, mTLS, or others.
+This is completely orthogonal to the use cases being tackled by this GEP.
+
+* The "mesh transport security" is something invisible to the user's application, and is simply used to secure communication between components in the mesh.
+* This proposal, instead, explicitly calls for sending TLS **to the user's application**.
+
+Whether the service mesh implementation uses mTLS or not, the mesh transport security is not relevant to this proposal.
+A service mesh using mTLS can *additionally* originate TLS that is terminated by the users application (that is, two layers of TLS!).
+
+![](images/mesh.png "Mesh transport")
+
+Possible future goals outside of the scope of this GEP include:
 
 1. Changes to the existing mechanisms for edge or passthrough TLS termination
-2. Providing a mechanism to decorate multiple route instances
-3. TLSRoute use cases
-4. UDPRoute use cases
-5. Controlling TLS versions or cipher suites used in TLS handshakes. (Use case #5 in [Gateway API TLS Use Cases](#references))
-6. Controlling certificates used by more than one workload (#6 in [Gateway API TLS Use Cases](#references))
-7. Client certificate settings used in TLS **from external clients to the
-Listener** (#7 in [Gateway API TLS Use Cases](#references))
-8. Providing a mechanism for the cluster operator to override gateway to backend TLS settings.
```

Comment:
Please don't remove.

Reply:
I think this has now effectively been split off into a separate GEP (https://github.com/kubernetes-sigs/gateway-api/pull/3876)

Reply:
Yes, GEP-3875 is now the gateway override for the config that's in this GEP.